client/homebrew/brewRenderer/toolBar/toolBar.jsx

```diff
@@ -0,0 +1,124 @@
+require('./toolBar.less');
+const React = require('react');
+const { useState, useEffect } = React;
+const _ = require('lodash')
+
+const MAX_ZOOM = 300;
+const MIN_ZOOM = 10;
+
+const ToolBar = ({ onZoomChange, currentPage, onPageChange, totalPages })=>{
+
+	const [zoomLevel, setZoomLevel] = useState(100);
+	const [pageInput, setPageInput] = useState(currentPage);
+
+	useEffect(()=>{
+		onZoomChange(zoomLevel);
+	}, [zoomLevel]);
+
+	useEffect(()=>{
+		setPageInput(currentPage);
+	}, [currentPage])
+
+	const handleZoomChange = (delta)=>{
+		const zoomChange = _.clamp(zoomLevel + delta, MIN_ZOOM, MAX_ZOOM);
+
+		setZoomLevel(zoomChange);
+	};
+
+	const handlePageChange = (page)=>{
+		setPageInput(page);
+	};
```

Comment:
Can we just use `setPageInput` directly instead of having this separate handler? (similar to how we use `setZoomLevel` directly on the zoom range slider.

Reply:
yup, done and done.  cls

Reply:
Actually, now addressing the input of non-integer values going to NaN, i think this handler method is an old vestige of preventing that issue that at some point was mistakenly stripped down.

So i'm likely going to bring this handler back, to check the input value before running parseInt on the input value.